mod-source-record-manager-server/src/main/java/org/folio/services/EventDrivenChunkProcessingServiceImpl.java

```diff
@@ -0,0 +1,170 @@
+import org.folio.dataimport.util.OkapiConnectionParams;
+import org.folio.rest.client.DataImportProfilesClient;
+import org.folio.rest.jaxrs.model.Event;
+import org.folio.rest.jaxrs.model.EventMetadata;
+import org.folio.rest.jaxrs.model.JobExecution;
+import org.folio.rest.jaxrs.model.JobExecutionSourceChunk;
+import org.folio.rest.jaxrs.model.RawRecordsDto;
+import org.folio.rest.jaxrs.model.Record;
+import org.folio.rest.jaxrs.model.StatusDto;
+import org.folio.rest.tools.PomReader;
+import org.folio.util.pubsub.PubSubClientUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import javax.ws.rs.NotFoundException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+import static java.lang.String.format;
+import static org.folio.HttpStatus.HTTP_OK;
+import static org.folio.rest.jaxrs.model.DataImportEventTypes.CREATED_SRS_MARC_BIB_RECORD;
+import static org.folio.rest.jaxrs.model.EntityType.MARC_BIBLIOGRAPHIC;
+
+@Service("eventDrivenChunkProcessingService")
+public class EventDrivenChunkProcessingServiceImpl extends AbstractChunkProcessingService {
+  private static final Logger LOGGER = LoggerFactory.getLogger(EventDrivenChunkProcessingServiceImpl.class);
+
+  private ChangeEngineService changeEngineService;
+
+  public EventDrivenChunkProcessingServiceImpl(@Autowired JobExecutionSourceChunkDao jobExecutionSourceChunkDao,
+                                               @Autowired JobExecutionService jobExecutionService,
+                                               @Autowired ChangeEngineService changeEngineService) {
+    super(jobExecutionSourceChunkDao, jobExecutionService);
+    this.changeEngineService = changeEngineService;
+  }
+
+  @Override
+  protected Future<Boolean> processRawRecordsChunk(RawRecordsDto incomingChunk, JobExecutionSourceChunk sourceChunk, String jobExecutionId, OkapiConnectionParams params) {
+    return checkAndUpdateJobExecutionStatusIfNecessary(jobExecutionId, new StatusDto().withStatus(StatusDto.Status.PARSING_IN_PROGRESS), params)
+      .compose(jobExecution -> changeEngineService.parseRawRecordsChunkForJobExecution(incomingChunk, jobExecution, sourceChunk.getId(), params))
+      .compose(records -> sendEventsWithCreatedRecords(records, jobExecutionId, params));
+  }
+
+  /**
+   * Sends events with created records to the mod-pubsub
+   *
+   * @param createdRecords records to send
+   * @param jobExecutionId job execution id
+   * @param params         connection parameters
+   * @return future with boolean
+   */
+  private Future<Boolean> sendEventsWithCreatedRecords(List<Record> createdRecords, String jobExecutionId, OkapiConnectionParams params) {
+    return jobExecutionService.getJobExecutionById(jobExecutionId, params.getTenantId())
+      .compose(jobOptional -> jobOptional
+        .map(jobExecution -> getJobProfileSnapshotWrapper(jobExecution, params))
```

Comment:
Let's get JobProfileWrapper once and store it in JobExecution to eliminate unnecessary calls to mod-data-import-converter-storage on every chunk.